src/sys/resource.rs

```diff
@@ -0,0 +1,230 @@
+//! Configure the process resource limits.
+use cfg_if::cfg_if;
+
+use crate::errno::Errno;
+use crate::Result;
+
+pub use libc::rlim_t;
+
+cfg_if! {
+    if #[cfg(all(target_os = "linux", target_env = "gnu"))]{
+        use libc::{__rlimit_resource_t, rlimit, RLIM_INFINITY};
+        libc_enum!{
+            #[repr(u32)]
+            pub enum Resource {
+                /// See detail of each Resource https://man7.org/linux/man-pages/man2/getrlimit.2.html
+                RLIMIT_AS,
+                RLIMIT_CORE,
+                RLIMIT_CPU,
+                RLIMIT_DATA,
+                RLIMIT_FSIZE,
+                RLIMIT_LOCKS,
+                RLIMIT_MEMLOCK,
+                RLIMIT_MSGQUEUE,
+                RLIMIT_NICE,
+                RLIMIT_NOFILE,
+                RLIMIT_NPROC,
+                RLIMIT_RSS,
+                RLIMIT_RTPRIO,
+                RLIMIT_RTTIME,
+                RLIMIT_SIGPENDING,
+                RLIMIT_STACK,
+            }
+        }
+    }else if #[cfg(any(
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "netbsd",
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "android",
+        target_os = "dragonfly",
+        target_os = "bitrig",
+        target_os = "linux", // target_env != "gnu"
+    ))]{
+        use libc::{c_int, rlimit, RLIM_INFINITY};
+
+        libc_enum! {
+            #[repr(i32)]
+            pub enum Resource {
+                /// See detail of each Resource https://man7.org/linux/man-pages/man2/getrlimit.2.html
```

Comment:
Hmm. Good question. The original idea was because I think the man page was a good reference that most are similar to this section. And it is easy to find(apologize for my lazyness). But I totoally agree that you are right. This link will be removed since it is for linux.